ast.c

```diff
@@ -14,6 +14,29 @@
 #include "zend_exceptions.h"
 #include "zend_smart_str.h"
 
+#ifndef ZEND_ARG_INFO_WITH_DEFAULT_VALUE
+#define ZEND_ARG_INFO_WITH_DEFAULT_VALUE(pass_by_ref, name, default_value) \
+	ZEND_ARG_INFO(pass_by_ref, name)
+#endif
+#if PHP_VERSION_ID < 70200
+#undef ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX
```

Comment:
Why do we need to undef something?

Reply:
The number of arguments accepted by that macro changed between php 7.1 and 7.2. Without redefining the macro, the arg info wouldn't compile.